src/splitwise.js

```diff
@@ -645,8 +709,16 @@ class Splitwise {
     R.values(METHODS).forEach((method) => {
       this[method.methodName] = generateEndpointMethod(method);
     });
-
-    this.getAccessToken = () => accessTokenPromise;
+    if (useOauth2) {
+      this.getAccessToken = (code, state) => {
+        if (!verifyState(state))
+          return Promise.reject(`State verification failed: ${state}`);
+        this.authCode = code;
+        return getAccessTokenFromAuthCode();
+      }
+    }
+    else 
+      this.getAccessToken = () => accessTokenPromise;
```

Comment:
would this still work sinceit's not an IIFE anymore?

Reply:
Yes this works perfectly